components/dashboards-web-component/src/DashboardCreatePage.jsx

```diff
@@ -0,0 +1,177 @@
+                    <label className="create-dashboard-page-label-name">
+                        Name of your Dashboard <span class="required">*</span>
+                    </label>
+                    <TextField onChange={this.handleURL} id="dashboardName" hintStyle={hintStyle}
+                               textareaStyle={textareaStyle}
+                               underlineStyle={underlineStyle} style={textFieldStyle}
+                               hintText="E.g. Sales Statistics"/>
+                </div>
+                <div class="form-group" className="create-dashboard-form-group">
+                    <label className="create-dashboard-page-label-url">
+                        URL <span class="required">*</span>
+                    </label>
+                    <TextField onChange={this.handleURL} value={this.state.url} id="dashboardURL" hintStyle={hintStyle}
+                               textareaStyle={textareaStyle} underlineStyle={underlineStyle} style={textFieldStyle}
+                               hintText="E.g. sales-stats"/>
+                </div>
+                <div class="form-group" className="create-dashboard-form-group">
+                    <label className="create-dashboard-page-label-description">
+                        Description <span class="required">*</span>
+                    </label>
+                    <TextField id="dashboardDescription" hintStyle={hintStyle} textareaStyle={textareaStyle}
+                               underlineStyle={underlineStyle} style={textFieldStyle}
+                               hintText="E.g. Monthly Sales Statistics"/>
+                </div>
+                <RaisedButton onClick={this.addDashboard} label="Add" primary={true}
+                              className="create-dashboard-button"/>
+                <Snackbar contentStyle={messageBoxStyle} bodyStyle={this.state.messageStyle} open={this.state.showMsg}
+                          message={this.state.Message} autoHideDuration={4000}
+                          onRequestClose={this.handleRequestClose}/>
+            </div>
+        </MuiThemeProvider>;
+    }
+
+    handleURL(event, value) {
+        this.setState({
+            url: this.sanitizeInput(value.toLowerCase().replace(/\s+/g, ''))
+        });
+    }
+
+    sanitizeInput(input) {
+        return input.replace(/[^a-z0-9-\s]/gim, "");
+    };
+
+    handleRequestClose() {
+        this.setState({
+            showMsg: false,
+            Message: ""
+        });
+    }
+
+    showError(message) {
+        this.setState({
+            messageStyle: errorMessageStyle,
+            showMsg: true,
+            Message: message
+        });
+    }
+
+    showMessage(message) {
+        this.setState({
+            messageStyle: successMessageStyle,
+            showMsg: true,
+            Message: message
+        });
+    }
+
+    addDashboard() {
+        let dashboardName = document.getElementById("dashboardName").value;
```

Comment:
Let's use an encoder to avoid cross site scripting vulnerabilities

Reply:
Since we are using react, we need to check that whether this is applicable or not.